New configuration tool

jasonleenaylor opened the pull request:
Changes related to creating a new configuration tool for the windows installers that doesn't require us to have an old version of IE installed with ActiveX controls enabled.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/sillsdev/masterinstaller/1)

<!-- Reviewable:end -->

jasonleenaylor committed:
Add solution for Utils projects and update to current VS

jasonleenaylor committed:
Add project and test project for new configuration tool

* We need to replace the old configuration tool which runs only
  in Internet Explorer with ActiveX turned on.
* Also updated old project files to work with VisualStudio 2013

jasonleenaylor committed:
Start adding the model options and tests

* Add Basic Model Reading and writing tests
* Handle the AutoConfigure options

jasonleenaylor committed:
Add reading of General section from xml

* Fill out the reading of the General section
* Add tests for all general options used in the WordSurv xml

jasonleenaylor committed:
Add serialization for MasterInstaller Product configuration

jasonleenaylor committed:
Add a ByHand test to verify all existing configs can be read

* Add ability to read empty element configurations of integers
  as 0
* Add ByHand test and a test for empty adjust elements

papeh commented on Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs:
```diff
@@ -0,0 +1,158 @@
+			public string ProductsPath { get; set; }
+
+			[XmlElement(ElementName = "CDsPath")]
+			public string CdPath { get; set; }
```
`Cd` vs `CDs` (fine as long as it's intentional)

papeh commented:
Reviewed 33 of 33 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

<sup>**[Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs, line 106 [r1]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K28h0hcA8TbgNAAlC9J)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/32f24fec1f45f03313f1d9a76eb5eec56477f1fd/Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs#L106)):</sup>
I may have made both ...NeedsShiftCtrl

<!-- Reviewable comment -K28h0hdpIKuM-TrMDb3:1900136937 -->

---

<sup>**[Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs, line 119 [r1]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K28ievXO49jFsmMZvCn)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/32f24fec1f45f03313f1d9a76eb5eec56477f1fd/Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs#L119)):</sup>
On line 35, the XmlElementName is `TermsOfUse`, but the Property name is`Terms`; here, the opposite

<!-- Reviewable comment -K28ievXO49jFsmMZvCo:633752080 -->

---

<sup>**[Code and Projects/MasterInstallerConfigurator/Properties/AssemblyInfo.cs, line 13 [r1]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K28jk-sreLj6qf2XlqG)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/32f24fec1f45f03313f1d9a76eb5eec56477f1fd/Code and Projects/MasterInstallerConfigurator/Properties/AssemblyInfo.cs#L13)):</sup>
SIL International?

<!-- Reviewable comment -K28jk-sreLj6qf2XlqH:-1739410944 -->

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1)

<!-- Sent from Reviewable.io -->

jasonleenaylor committed:
Initial work toward the View components

I started to work with the controller and view and realized
the old tool was using javascript to store settings in addition
to the xml file.

jasonleenaylor committed:
Start converting .js settings into xml model settings

* Convert flavor name and url settings

jasonleenaylor committed:
Add handling of the products selected for each flavor

* Save the selected product titles for each flavor
* Fix one model property name inconsistancy uncovered in
  review

jasonleenaylor commented:
<sup>**[Code and Projects/MasterInstallerConfigurator/Properties/AssemblyInfo.cs, line 13 [r1]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K28jk-sreLj6qf2XlqG)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/32f24fec1f45f03313f1d9a76eb5eec56477f1fd/Code and Projects/MasterInstallerConfigurator/Properties/AssemblyInfo.cs#L13)):</sup>
Yes, I need to do this in a further commit.

<!-- Reviewable comment -K2Cow0Cyp2jhMSMMHtS:1942284886 -->

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1)

<!-- Sent from Reviewable.io -->

papeh commented:
Reviewed 10 of 10 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

<sup>**[Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs, line 194 [r3]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K2ClZ-Sh-PHoCT8rUEk)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/8627b37478b2aa1c585beba423726f7067a7e16a/Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs#L194)):</sup>
Theoretically, a client could set ValueString to a non-int; however, I'm not sure how much we need to worry about this problem in this project.

<!-- Reviewable comment -K2ClZ-Sh-PHoCT8rUEl:338493467 -->

---

<sup>**[Code and Projects/MasterInstallerConfigurator/JavaScriptConverter.cs, line 99 [r3]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K2Cr1WiSWhawFiQONc0)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/8627b37478b2aa1c585beba423726f7067a7e16a/Code and Projects/MasterInstallerConfigurator/JavaScriptConverter.cs#L99)):</sup>
I assume this will be filled in later?

<!-- Reviewable comment -K2Cr1WiSWhawFiQONc1:1313494473 -->

---

<sup>**[Code and Projects/MasterInstallerConfiguratorTests/MasterInstallerConfiguratorTests/JavaScriptConverterTests.cs, line 86 [r3]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K2Cqb41CofRFo0fgs2N)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/8627b37478b2aa1c585beba423726f7067a7e16a/Code and Projects/MasterInstallerConfiguratorTests/MasterInstallerConfiguratorTests/JavaScriptConverterTests.cs#L86)):</sup>
I don't see where these true and false values are used

<!-- Reviewable comment -K2Cqb42I8oo4E4dD05M:-1332823394 -->

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1)

<!-- Sent from Reviewable.io -->

jasonleenaylor commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

<sup>**[Code and Projects/MasterInstallerConfigurator/JavaScriptConverter.cs, line 99 [r3]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K2Cr1WiSWhawFiQONc0)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/8627b37478b2aa1c585beba423726f7067a7e16a/Code and Projects/MasterInstallerConfigurator/JavaScriptConverter.cs#L99)):</sup>
Yes.

<!-- Reviewable comment -K2DIdtLCjUn_Zl6A20z:677251276 -->

---

<sup>**[Code and Projects/MasterInstallerConfiguratorTests/MasterInstallerConfiguratorTests/JavaScriptConverterTests.cs, line 86 [r3]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K2Cqb41CofRFo0fgs2N)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/8627b37478b2aa1c585beba423726f7067a7e16a/Code and Projects/MasterInstallerConfiguratorTests/MasterInstallerConfiguratorTests/JavaScriptConverterTests.cs#L86)):</sup>
In the FlavorProducts line 75

<!-- Reviewable comment -K2DIyNP9u00U1zyNlF5:-1377010671 -->

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1)

<!-- Sent from Reviewable.io -->
